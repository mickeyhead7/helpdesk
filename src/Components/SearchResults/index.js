import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import BodyHeader from '@Components/BodyHeader';
import ArticleList from '@Components/ArticleList';
import I18nProvider from '@packages/Components/i18n/Provider';
import Translate from '@packages/Components/i18n/Translate';
import enGB from './languages/en-GB';

const phrases = {
  'en-GB': enGB,
};

export const SearchResults = ({
  articles,
  locale,
  search,
}) => (
  <I18nProvider locale={locale} phrases={phrases}>
    <BodyHeader>
      <h1>
        <Translate locale={locale} phrase="messages.searchResultsFor" params={{ search }} />
      </h1>
    </BodyHeader>
    <ArticleList articles={articles} />
  </I18nProvider>
);

SearchResults.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    node: PropTypes.shape({
      frontmatter: PropTypes.shape({
        excerpt: PropTypes.string.isRequired,
        path: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  })).isRequired,
  locale: PropTypes.string.isRequired,
  search: PropTypes.string.isRequired,
};

const mapStateToProps = ({ i18n }) => ({
  locale: i18n.locale,
});

export default connect(mapStateToProps)(SearchResults);
