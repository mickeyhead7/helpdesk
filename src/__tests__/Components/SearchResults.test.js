import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SearchResults } from '@Components/SearchResults';

const articles = [];

test('SearchResults', () => {
  const props = {
    locale: 'en-GB',
    articles,
    search: 'search-text',
  };
  const component = shallow(<SearchResults {...props} />);

  expect(toJson(component)).toMatchSnapshot();
});
