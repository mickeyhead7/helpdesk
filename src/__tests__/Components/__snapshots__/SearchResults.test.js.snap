// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults 1`] = `
<Connect(Provider)
  locale="en-GB"
  phrases={
    Object {
      "en-GB": Object {
        "messages": Object {
          "searchResultsFor": "Search results for \\"%{search}\\"",
        },
      },
    }
  }
>
  <BodyHeader>
    <h1>
      <Connect(Translate)
        locale="en-GB"
        params={
          Object {
            "search": "search-text",
          }
        }
        phrase="messages.searchResultsFor"
      />
    </h1>
  </BodyHeader>
  <ArticleList
    articles={Array []}
  />
</Connect(Provider)>
`;
